pkg/resource/interfaces.go

```diff
@@ -192,14 +195,21 @@ type Object interface {
 // resource (e.g. a CloudSQL instance).
 type Managed interface { //nolint:interfacebloat // This interface has to be big.
 	Object
+	Manageable
+	Conditioned
+}
 
-	ProviderConfigReferencer
+type ModernManaged interface {
+	Managed
+	LocalConnectionSecretWriterTo
+	TypedProviderConfigReferencer
+}
+
+type LegacyManaged interface {
+	Managed
 	ConnectionSecretWriterTo
-	ConnectionDetailsPublisherTo
-	Manageable
+	ProviderConfigReferencer
 	Orphanable
-
-	Conditioned
 }
```

Comment:
Naming nit: I don't love `ModernManaged` but I also can't think of a better alternative. Only thing that comes to mind would be:

* `BaseManaged`
* `Managed`
* `LegacyManaged`

Reply:
FWIW I've warmed to `ModernManaged` and `LegacyManaged`. It's pretty close to what we did with XR - i.e. they have `SchemaModern` and `SchemaLegacy`.